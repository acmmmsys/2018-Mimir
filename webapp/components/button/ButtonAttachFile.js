import React from 'react';
import { connect } from 'react-redux';
import { compose, withProps, withHandlers } from 'recompose';
import { attachFile } from 'actions';

import { isEmpty } from 'utils';

import ButtonAction from 'components/button/ButtonAction'

const enhance = compose(
    connect(
        null,
        dispatch => ({
            attachImage(imageId) {
                dispatch(attachFile(imageId))
            },
        }),
    ),
    withProps(props => ({
        label: 'Attach Image',
    })),
    withHandlers({
        onClick: props => event => {
            if (props.imageId) {
                props.attachImage(props.imageId);
            }
        }
    })
);

export default enhance(ButtonAction);