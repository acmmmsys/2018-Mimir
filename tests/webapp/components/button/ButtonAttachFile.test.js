import React from 'react';
import ButtonAttachFile from '../../../../webapp/components/button/ButtonAttachFile'
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store'

const mockStore = configureStore();

let store;
let wrapper;

test('ButtonAttachFile does render', () => {
  const wrapper = shallow(
      <ButtonAttachFile store={mockStore()}>This is a test!</ButtonAttachFile>
  );
  expect(wrapper).toMatchSnapshot();
});
