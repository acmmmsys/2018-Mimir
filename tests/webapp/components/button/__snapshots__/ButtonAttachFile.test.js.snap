// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonAttachFile does render 1`] = `
<withProps(withHandlers(ButtonAction))
  attachImage={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  This is a test!
</withProps(withHandlers(ButtonAction))>
`;
